cmake/Packaging.cmake

```diff
@@ -94,6 +94,22 @@ endif()
 # cpack generation. CPACK_BINARY_DIR persists across cpack invocations
 set(LY_CMAKE_PACKAGE_DOWNLOAD_PATH ${CPACK_BINARY_DIR}/${CPACK_CMAKE_PACKAGE_FILE})
 
+# Scan the source and packages for licenses, then add the generated license results to the binary output folder
+set(CPACK_3P_LICENSE_FILE "${CPACK_BINARY_DIR}/NOTICES.txt")
+set(CPACK_3P_MANIFEST_FILE "${CPACK_BINARY_DIR}/SPDX-License.json")
+
+configure_file(${LY_ROOT_FOLDER}/cmake/Packaging/LicenseScan.cmake.in
+    ${CPACK_BINARY_DIR}/LicenseScan.cmake
+    @ONLY
+)
+ly_install(SCRIPT ${CPACK_BINARY_DIR}/LicenseScan.cmake
+    COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
+)
+ly_install(FILES ${CPACK_3P_LICENSE_FILE} ${CPACK_3P_MANIFEST_FILE}
+    DESTINATION .
+    COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
+)
```

Comment:
is this meant go into the installer, next to the installer or both? (noticing in other files that the license and manifest files are also uploaded to S3

Reply:
Both, for a few reasons:

1. The notices file gets to live in the installed folder, so that there's an offline version (the installer also has a direct link to the one in https://github.com/o3de/3p-package-source during the bootstrap)

2. When we generate the notices file, it will be specific to the version and platform it's generated for (since 3p differs for each). This will be convenient for the user to pull and for automation to update https://github.com/o3de/3p-package-source, whenever main gets a new tag